src/core/io/src/4C_io_input_spec_builders.hpp

```diff
@@ -1392,6 +1404,45 @@ namespace Core::IO
     [[nodiscard]] InputSpec selection(std::string name, Internal::BasedOn<Selector> based_on,
         SelectionData<StorageType> data = {});
 
+    /**
+     * Defines an input field for spatially dependent parameters.
+     *
+     * The input field allows a parameter of a certain type to be specified as a function of element
+     * or node IDs, enabling spatially varying input data. This is useful for cases where a
+     * parameter, such as material stiffness, needs to be defined differently for different parts of
+     * a model.
+     *
+     * Example usage:
+     * @code
+     * auto spec = input_field<double>("stiffness");
+     * @endcode
+     *
+     * This will match the following input:
+     * @code
+     *  stiffness:
+     *    type: constant
+     *    value: 100.0
+     * @endcode
+     *  or
+     * @code
+     *  stiffness:
+     *    type: from_file
+     *    value: /some/file.json
+     * @endcode
+     * The json input field json file could look like this:
+     * @code
+     * {
+     *  "stiffness": {
+     *    "1": 2.0,
+     *    "2": 3.5,
+     *    "3": 4.0,
+     *    "4": 5.5
+     *    }
+     * }
+     */
+    template <typename T>
+    [[nodiscard]] InputSpec input_field(std::string name, ParameterDataIn<T> data = {});
```

Comment:
I like the docs!

Not sure this function should be inside this header since it is a special case. Currently, we do not have another header for such specs, though. I'll keep it in mind and we can move it later.